packages/sirius-web/backend/sirius-web-services/src/main/java/org/eclipse/sirius/web/services/explorer/ExplorerInitialDirectEditTreeItemLabelProvider.java

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.services.explorer;
+
+import java.util.Optional;
+
+import org.eclipse.sirius.components.collaborative.trees.api.IInitialDirectEditTreeItemLabelProvider;
+import org.eclipse.sirius.components.collaborative.trees.dto.InitialDirectEditElementLabelInput;
+import org.eclipse.sirius.components.collaborative.trees.dto.InitialDirectEditElementLabelSuccessPayload;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IPayload;
+import org.eclipse.sirius.components.trees.Tree;
+import org.eclipse.sirius.components.trees.TreeItem;
+import org.springframework.stereotype.Service;
+
+/**
+ * ExplorerInitialDirectEditTreeItemLabelProvider.
```

Comment:
Could you provide a Javadoc with something a tiny bit more useful than just the name of the class (especially since that may change over time). Something like `Used to compute the initial label to display when the direct edit of a tree item of the explorer is triggered on the frontend`